### ISSUE TITLE
Problem when receiving an incoming message with noAuthNoPriv security level

### ISSUE THREAD

**Issue by LenardJensen** [lextudio/pysnmp#41]
Hello,
we encountered a problem with the latest version of pysnmp-lextudio in which whe hoped you might be able to help us.

When we are trying to connect with a client using SNMP with an authentication that should result in an UnknownUserName error, we instead run into a loop of retries, as the client will continue to send `(1, 3, 6, 1, 6, 3, 15, 1, 1, 3, 0): errind.unknownUserName` responses that are seemingly dismissed by pysnmp.
During an investigation it seems like this commit was the cause of this new behaviour: [Fixed #6 by enabling time data in noAuthNoPriv](https://github.com/lextudio/pysnmp/commit/e960bf042c059d81694f6d4bcaf17ef0819703e1#diff-e369b3ed5bd7bcb6b704662f4b9ab8e54ba482d45cf4d1b287b2c629046b74daR896)
Previously an incoming message with noAuthNoPriv set as its security level resulted in reading out the time data without any extra authentication, however now an incoming message will trigger the `authHandler.authenticateIncomingMsg` call, which will result in an `errind.authenticationFailure` status being raised. Was this an intended change to still authenticate incoming messages with a security level of 1, as long as an user name was set? 
Changing the code back to exclude a security level of 1 from triggering the authentication method will again result in the old and expected behaviour with no other changes needed.

Pysnmp is currently wrapped in another high level library that abstracts the whole SNMP connection, so it may be possible that we've just encountered a breaking change and need to catch the authentication error somewhere. However currently it appears as if this specific error gets handled inside pysnmp, causing the connection to rather time out instead of terminating it on our end by receiving the UnknownUserName error.

If additional infos are needed, I'd be happy to provide it, but both Wireshark protocols and debug logs may contain sensitive information that we have to check before posting them here.

Regards

**Comment by lextm** [comment]
Please write to support@lextudio.com to open a free support ticket.

**Comment by lextm** [comment]
We finished initial investigation and don't think there is anything wrong with the current code base.

## NET-SNMP Behavior
First we check what de facto tool NET-SNMP works in such a case,

``` bash
~ snmpget -v3 -u usr-none-none-not-exist -l noAuthNoPriv -r 0 -t 1 demo.pysnmp.com SNMPv2-MIB::sysDescr.0
No log handling enabled - using stderr logging
snmpget: Decryption error
```
## PySNMP Behavior
Convert the NET-SNMP command to equivalent code with PySNMP as below and we get a test case,

``` python
import pytest
from pysnmp.hlapi.asyncio import *
from pysnmp.proto.errind import DecryptionError

@pytest.mark.asyncio
async def test_usm_no_auth_no_priv_wrong_user():
    snmpEngine = SnmpEngine()
    authData = UsmUserData(
        "usr-none-none-not-exist"
    )
    errorIndication, errorStatus, errorIndex, varBinds = await getCmd(
        snmpEngine,
        authData,
        UdpTransportTarget(("demo.pysnmp.com", 161), retries=0),
        ContextData(),
        ObjectType(ObjectIdentity("SNMPv2-MIB", "sysDescr", 0)),
    )

    assert isinstance(errorIndication, DecryptionError)
    # assert str(errorIndication) == 'Ciphering services not available or ciphertext is broken'
```

# Comments

* By capturing the packets in both cases we can see the same conversation pattern is there.
* We made significant changes in this fork to eliminate incompatibility with other SNMP implementations such as #SNMP, and many such cases were verified. More thorough test cases will be added in the coming releases to increase unit test coverage.

**Comment by lextm** [comment]
While we added more test cases to ensure v3 support is compatible with other SNMP implementations, another issue was found and fixed in 5.0.34 release.

Thus, you might want to upgrade to this new release and see if any issue remains for your scenario.

**Comment by LenardJensen** [comment]
Thank you very much for investigating the issue, the problem is now fixed in the 5.0.34 release
